cpp/include/cudf/groupby.hpp

```diff
@@ -195,6 +209,41 @@ class groupby {
   groups get_groups(cudf::table_view values             = {},
                     rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+  /**
+   * @brief Performs grouped replace nulls on @p value
+   *
+   * For each `value[i] == NULL` in group `j`, `value[i]` is replaced with the first non-null value
+   * in group `j` that precedes or follows `value[i]`. If a non-null value is not found in the
+   * specified direction, `value[i]` is left NULL.
+   *
+   * The returned pair contains a column of the sorted keys and a `scan_aggregation_result`.
+   *
+   * `scan_aggregation_result` contains the result column in which values of the same group is
+   * located in a contiguous memory.
+   *
+   * Example:
+   * @code{.pseudo}
+   *
+   * //Inputs:
+   * keys:    {2, 1, 2, 1}
+   * values:  {3, 4, NULL, NULL}
+   *
+   * //Outputs:
+   * sorted_keys:     {1, 1, 2, 2}
+   * result:          {4, 4, 3, 3}
+   * @endcode
+   *
+   * @param[in] value A column whose null values will be replaced.
+   * @param[in] replace_policy Specify the position of replacement values relative to null values.
+   * @param[in] mr Device memory resource used to allocate device memory of the returned column.
+   *
+   * @return Pair that contains a table with the sorted keys and a `scan_aggregation_result`
+   */
+  std::pair<std::unique_ptr<table>, scan_aggregation_result> replace_nulls(
```

Comment:
On further thought, what's the purpose of having `replace_nulls` return a `scan_aggregation_result` instead of just a `unique_ptr<column>`? We can never return more than a single column here yes?

Reply:
Ah yes, this is just a single aggregation request, shouldn't have used it here. Gonna remove it from this PR.